build/Dockerfile

```diff
@@ -1,6 +1,7 @@
 # Build the manager binary
-FROM golang:1.20 as builder
+FROM --platform=$BUILDPLATFORM golang:1.20 as builder
 
+ARG TARGETOS TARGETARCH
```

Comment:
nit
```suggestion
ARG TARGET_OS TARGET_ARCH
```

Reply:
Doesnt read TARGET_OS and TARGET_ARCH as arguments when using `--platform linux/amd64,...`